In [1]:
import pandas as pd

# Khám phá dữ liệu 1 

File dữ liệu store_info_fixed_29-9.txt

In [145]:
store_info_df = pd.read_csv('store_info_fixed_29-9.txt',delimiter=',',index_col='idNum')
store_info_df.head()

_id  \
idNum                             
0      63340c41b53fc66fec0031a6   
1      63340c43b53fc66fec0031a7   
2      63340c4db53fc66fec0031a8   
3      63340c50b53fc66fec0031a9   
4      63340c53b53fc66fec0031aa   

                                                    name review_count  \
idNum                                                                   
0                      Bún Đậu Homemade - Phan Xích Long          211   
1               Da Heo Út Cưng - Shop Online - Vĩnh Viễn            9   
2                              McDonald’s Vincom Thu Duc           16   
3      Crystal Jade Kitchen - Vạn Hạnh Mall - Nhà Hàn...           68   
4                 Kichi Kichi Lẩu Băng Chuyền - Cộng Hòa          113   

                                            address         district  \
idNum                                                                  
0                           25 Phan Xích Long, P. 3  Quận Bình Thạnh   
1                               208 Vĩnh Viễn, P. 4          Quận 10   
2      Vincom Thủ Đức, 216 Võ Văn Ngân, P. Bình Thọ      Tp. Thủ Đức   
3           06 Vạn Hạnh Mall, 11 Sư Vạn Hạnh, P. 12          Quận 10   
4                   Tầng Trệt, 15-17 Cộng Hòa, P. 4    Quận Tân Bình   

      average_score  views food_list comment_list  position_score  ...  \
idNum                                                              ...   
0               6.6  22.2K        []           []             7.4  ...   
1               7.0    652        []           []             7.4  ...   
2               7.7   2.8K        []           []             7.9  ...   
3               8.2   5.9K        []           []             8.3  ...   
4               7.5  28.0K        []           []             7.6  ...   

                     holiday         type       capacity  \
idNum                                                      
0      Các ngày lễ trong năm      Quán ăn   40 người lớn   
1                        NaN  Shop Online            NaN   
2      Các ngày lễ trong năm      Quán ăn   50 người lớn   
3      Các ngày lễ trong năm     Nhà hàng  100 người lớn   
4      Các ngày lễ trong năm     Nhà hàng  200 người lớn   

                        style  \
idNum                           
0                    Món Việt   
1                    Món Việt   
2               Mỹ ,  Châu Mỹ   
3      Món Á ,  Món Trung Hoa   
4           Món Á ,  Món Nhật   

                                           suitable_with  \
idNum                                                      
0                                  Ăn gia đình ,  Ăn vặt   
1                                                    NaN   
2      Ăn gia đình ,  Họp nhóm ,  Ăn vặt ,  Takeaway ...   
3      Đãi tiệc ,  Ăn gia đình ,  Họp nhóm ,  Tiếp khách   
4        Ăn gia đình ,  Hẹn hò ,  Họp nhóm ,  Tiếp khách   

                                                  serves  \
idNum                                                      
0                   Bún ,  Chè ,  Lẩu ,  Ăn vặt - Ăn nhẹ   
1                                                    NaN   
2        Gà rán ,  Hamburger ,  Fastfood - Thức ăn nhanh   
3      Cháo ,  Chân cánh gà nướng ,  Gà ác/gà hầm ,  ...   
4                                                    Lẩu   

                                               available  \
idNum                                                      
0      ['Có wifi', 'Có máy lạnh & điều hòa', 'Cho mua...   
1                      ['Có giao hàng', 'Nên đặt trước']   
2      ['Có wifi', 'Có máy lạnh & điều hòa', 'Cho mua...   
3                           ['Có wifi', 'Nên đặt trước']   
4      ['Có wifi', 'Có máy lạnh & điều hòa', 'Nên đặt...   

                                            parking_lots           website  \
idNum                                                                        
0      [{'name': 'Bãi Giữ Xe Công Viên Lê Văn Tám - V...               NaN   
1                                                     []               NaN   
2                 

## Thông tin chung về dữ liệu 

In [146]:
# Số dòng và cột
print('Số dòng và số cột: {0} {1}'.format(store_info_df.shape[0],store_info_df.shape[1]))
# Có dòng bị lặp không
print('Số dòng bị lặp: ', store_info_df.duplicated().sum())

Số dòng và số cột: 2092 29
Số dòng bị lặp:  0


Mỗi dòng trong dữ liệu có các thông tin về quán ăn như: Tên quán, địa chỉ, số điểm đánh giá theo tiêu chí, ...

Kiểu dữ liệu cho từng cột. Nhìn qua thì các điểm đánh giá là kiểu float64. Tuy nhiên một vài cột là kiểu dữ liệu dạng số nhưng hiện tại đang có kiểu dữ liệu là object như: review_count, average_score, views. Chúng ta cần lưu ý những cột này trong những bước sau. 

In [147]:
store_info_df.dtypes

_id                   object
name                  object
review_count          object
address               object
district              object
average_score         object
views                 object
food_list             object
comment_list          object
position_score       float64
space_score          float64
quality_score        float64
serve_score          float64
price_score          float64
food_link             object
opening_time          object
suitable_time         object
last_pick_up_time     object
prepare_time          object
holiday               object
type                  object
capacity              object
style                 object
suitable_with         object
serves                object
available             object
parking_lots          object
website               object
price_range           object
dtype: object

Kiểm tra thông tin của các cột review_count, average_score, views

In [148]:
store_info_df[['review_count', 'average_score', 'views']].describe()

review_count average_score views
count          2092          2092  2092
unique          466            73  1059
top               0           _._  1.3K
freq            160           160    18

In [149]:
store_info_df['review_count'].unique()

array(['211', '9', '16', '68', '113', '287', '1,365', '42', '22', '169',
       '483', '0', '30', '20', '160', '101', '213', '346', '17', '120',
       '11', '80', '1', '316', '187', '143', '272', '59', '153', '227',
       '142', '550', '206', '14', '152', '2', '167', '107', '215', '34',
       '217', '131', '67', '133', '75', '51', '46', '126', '123', '39',
       '297', '49', '158', '15', '135', '38', '175', '32', '36', '279',
       '197', '150', '255', '3', '522', '23', '397', '244', '31', '89',
       '61', '70', '867', '60', '571', '375', '424', '92', '258', '139',
       '344', '111', '148', '44', '81', '5', '532', '117', '13', '124',
       '548', '12', '29', '73', '21', '125', '114', '219', '754', '363',
       '164', '1,089', '41', '103', '199', '87', '154', '731', '168',
       '69', '24', '127', '179', '256', '115', '159', '183', '543', '162',
       '149', '116', '91', '415', '252', '163', '231', '7', '45', '109',
       '145', '253', '218', '40', '10', '90', '144', '141'

In [150]:
store_info_df['review_count'] = store_info_df['review_count'].str.replace(',','').astype('int32')
store_info_df['review_count'].unique()

array([ 211,    9,   16,   68,  113,  287, 1365,   42,   22,  169,  483,
          0,   30,   20,  160,  101,  213,  346,   17,  120,   11,   80,
          1,  316,  187,  143,  272,   59,  153,  227,  142,  550,  206,
         14,  152,    2,  167,  107,  215,   34,  217,  131,   67,  133,
         75,   51,   46,  126,  123,   39,  297,   49,  158,   15,  135,
         38,  175,   32,   36,  279,  197,  150,  255,    3,  522,   23,
        397,  244,   31,   89,   61,   70,  867,   60,  571,  375,  424,
         92,  258,  139,  344,  111,  148,   44,   81,    5,  532,  117,
         13,  124,  548,   12,   29,   73,   21,  125,  114,  219,  754,
        363,  164, 1089,   41,  103,  199,   87,  154,  731,  168,   69,
         24,  127,  179,  256,  115,  159,  183,  543,  162,  149,  116,
         91,  415,  252,  163,  231,    7,   45,  109,  145,  253,  218,
         40,   10,   90,  144,  141,   37,  957,   76,  299,    4,  324,
        100,  686,  172,  268,  250,  104,   33,  1

In [151]:
store_info_df['average_score'] = store_info_df['average_score'].str.replace('_._','nan',regex=False).astype('float32')
store_info_df['average_score'].unique()

array([ 6.6,  7. ,  7.7,  8.2,  7.5,  7.3,  5.8,  6.5,  9.1,  nan,  8.9,
        7.1,  8.6,  7.2,  5.9,  6.8,  6.4,  7.4,  8. ,  8.8,  6.9,  8.3,
        7.9,  5.5,  7.8,  7.6,  8.4,  6.3,  4.7,  4.8,  8.1,  6. ,  6.2,
        9.4,  4.5,  5. ,  6.7,  6.1,  9.9,  9.2,  9. ,  5.6,  8.5,  9.8,
        9.7,  4.6,  5.4,  3.6,  8.7, 10. ,  5.3,  4.9,  4. ,  9.5,  2.8,
        5.7,  1. ,  4.1,  5.2,  9.3,  1.6,  2. ,  3.3,  3.1,  4.2,  5.1,
        9.6,  3.8,  4.3,  2.4,  3.4,  2.2,  3.2], dtype=float32)

In [152]:
a = store_info_df['views'].str.extract(r'(\d+.\d+)K')
b = store_info_df['views'].str.extract(r'(\d+)')
a = a + '00'
a = a.fillna(b)
store_info_df['views'] = a[0].str.replace('.','',regex=False)
store_info_df['views'] = store_info_df['views'].astype('int64')

Kiểm tra lại thông tin của 3 cột review_count, average_score, views

In [153]:
 store_info_df[['review_count', 'average_score', 'views']].describe()

review_count  average_score          views
count   2092.000000    1932.000000    2092.000000
mean     133.703633       7.184006   35921.063576
std      166.472497       0.998392   49540.589209
min        0.000000       1.000000       7.000000
25%       19.000000       6.700000    2800.000000
50%       89.000000       7.200000   12800.000000
75%      179.000000       7.625000   55800.000000
max     1513.000000      10.000000  459400.000000

- Các cột dữ liệu dạng số là: 

` 'review_count', 'average_score', 'space_score', 'position_score', 'quality_score','serve_score','price_score','views'`

- Các cột dữ liệu dạng phân loại là:

` '_id', 'name', 'address', 'district', 'opening_time', 'suitable_time', 'last_pick_up_time',  'prepare_time', 'holiday', 'type', 'style', 'serves', 'suitable_with', 'available', 'capacity', 'serves',  'price_range' `

- Các cột dữ liệu còn lại:
` 'food_list', 'comment_list', 'food_link', 'parking_lots', 'website'  ` (Chúng ta sẽ xử lý những dữ liệu này khi cần đến)

Kiểm tra lại thông tin của những cột dữ liệu trên

In [154]:
store_info_df[\
['review_count', 'average_score', 'space_score', 'position_score', 'quality_score','serve_score','price_score','views']]\
.describe()

review_count  average_score  space_score  position_score  \
count   2092.000000    1932.000000  2092.000000     2092.000000   
mean     133.703633       7.184006     6.574379        6.765918   
std      166.472497       0.998392     2.418762        2.413742   
min        0.000000       1.000000    -1.000000       -1.000000   
25%       19.000000       6.700000     6.300000        6.800000   
50%       89.000000       7.200000     7.100000        7.300000   
75%      179.000000       7.625000     7.800000        7.800000   
max     1513.000000      10.000000    10.000000       10.000000   

       quality_score  serve_score  price_score          views  
count    2092.000000  2092.000000  2092.000000    2092.000000  
mean        6.673996     6.444503     6.342495   35921.063576  
std         2.474057     2.433554     2.339672   49540.589209  
min        -1.000000    -1.000000    -1.000000       7.000000  
25%         6.600000     6.200000     6.300000    2800.000000  
50%         7.300000     7.000000     6.800000   12800.000000  
75%         7.900000     7.600000     7.300000   55800.000000  
max        10.000000    10.000000    10.000000  459400.000000

Nhìn qua thì cột điểm đánh giá như ` 'space_score', 'position_score', 'quality_score', 'serve_score', 'price_score' ` có điểm đánh giá nhỏ nhất là -1. Chúng ta sẽ nhìn qua những dòng này.

In [155]:
pd.set_option('display.max_colwidth', None)
store_info_df[(store_info_df['space_score'] == -1) |\
             (store_info_df['position_score'] == -1) |\
             (store_info_df['quality_score'] == -1) |\
             (store_info_df['serve_score'] == -1) |\
             (store_info_df['price_score'] == -1)][['name','food_link','space_score',\
                                                    'position_score', 'quality_score', 'serve_score', 'price_score']]

name  \
idNum                                                               
11                                Bánh Bao Thọ Phát - Lâm Văn Bền   
13     FoodFest - Con Gà Mái - Cơm Gà Phú Yên - NowFoodxCoca-Cola   
14                        Kem Baskin Robbins - Vincom Landmark 81   
17                            Nhà Hàng Cơm Niêu Sài Gòn - Pasteur   
19                           Steak Bin & Pizza - Trương Công Định   
...                                                           ...   
2040                                Steak Bin & Pizza - Phổ Quang   
2048                      Trà Sữa ChaGo Tea & Cafe - Tôn Thất Đạm   
2060                                   TL Sushi & Sashimi - Tô Ký   
2076                              Remnant Coffee - Huỳnh Tấn Phát   
2078                    Chickita - Gà Nướng Lửa Hồng - Sky Garden   

                                                                                                  food_link  \
idNum                                                                                                         
11             https://shopeefood.vn/ho-chi-minh/banh-bao-tho-phat-lam-van-ben?source_url=foody_ordernow_pc   
13                                                                                                      NaN   
14     https://shopeefood.vn/ho-chi-minh/kem-baskin-robbins-vincom-landmark-81?source_url=foody_ordernow_pc   
17                                                                                                      NaN   
19          https://shopeefood.vn/ho-chi-minh/steak-bin-pizza-truong-cong-dinh?source_url=foody_ordernow_pc   
...                                                                                                     ...   
2040               https://shopeefood.vn/ho-chi-minh/steak-bin-pizza-pho-quang?source_url=foody_ordernow_pc   
2048     https://shopeefood.vn/ho-chi-minh/tra-sua-chago-tea-cafe-ton-that-dam?source_url=foody_ordernow_pc   
2060                  https://shopeefood.vn/ho-chi-minh/tl-sushi-sashimi-to-ky?source_url=foody_ordernow_pc   
2076                                                                                                    NaN   
2078   https://shopeefood.vn/ho-chi-minh/chickita-ga-nuong-lua-hong-sky-garden?source_url=foody_ordernow_pc   

       space_score  position_score  quality_score  serve_score  price_score  
idNum                                                                        
11            -1.0            -1.0           -1.0         -1.0         -1.0  
13            -1.0            -1.0           -1.0         -1.0         -1.0  
14            -1.0            -1.0           -1.0         -1.0         -1.0  
17            -1.0            -1.0           -1.0         -1.0         -1.0  
19            -1.0            -1.0           -1.0         -1.0         -1.0  
...            ...             ...            ...          ...          ...  
2040          -1.0            -1.0           -1.0         -1.0         -1.0  
2048          -1.0            -1.0           -1.0         -1.0         -1.0  
2060          -1.0            -1.0           -1.0         -1.0         -1.0  
2076          -1.0            -1.0           -1.0         -1.0         -1.0  
2078          -1.0            -1.0           -1.0         -1.0         -1.0  

[160 rows x 7 columns]

Có 160 dòng có những giá trị -1 như thế này.

Qua kiểm tra địa chỉ của những quán này trên shopee thì những giá trị -1 này đại diện cho giá trị rỗng.

In [156]:
store_info_df[['name', 'address', 'district', 'opening_time', 'suitable_time',\
    'last_pick_up_time',  'prepare_time', 'holiday', 'type', 'style', 'serves',\
    'suitable_with', 'available','capacity', 'serves',  'price_range']].describe()

name  \
count                               2092   
unique                              2088   
top     TukTuk Thai Bistro - Lý Tự Trọng   
freq                                   2   

                                                    address district  \
count                                                  2092     2092   
unique                                                 2011       23   
top     Tầng B3 Vincom Center, 72 Lê Thánh Tôn, P. Bến Nghé   Quận 1   
freq                                                      7      513   

                      opening_time  \
count                         2092   
unique                         396   
top     Đang mở cửa  10:00 - 22:00   
freq                           245   

                                     suitable_time last_pick_up_time  \
count                                         2088              1233   
unique                                          22                73   
top     Buổi sáng , Buổi trưa , Buổi xế , Buổi tối          09:30 PM   
freq                                          1241               445   

              prepare_time                holiday     type     style  \
count                 1888                   1888     2092      1918   
unique                  45                     45       62        97   
top     Khoảng 5 - 10 phút  Các ngày lễ trong năm  Quán ăn  Món Việt   
freq                   767                   1752      566       830   

                                    serves  \
count                                 1970   
unique                                1103   
top     Trà ,  Thức uống ,  Trà sữa ,  Sữa   
freq                                    48   

                                   suitable_with available      capacity  \
count                                       1889      2092          1876   
unique                                       540       551            67   
top     Hẹn hò ,  Họp nhóm ,  Takeaway - Mang về        []  50 người lớn   
freq                                         144       165           372   

                                    serves           price_range  
count                                 1970                   132  
unique                                1103                   103  
top     Trà ,  Thức uống ,  Trà sữa ,  Sữa  50,000 đ - 200,000 đ  
freq                                    48                     5

In [157]:
store_info_df[['name', 'address', 'district', 'opening_time', 'suitable_time',\
    'last_pick_up_time',  'prepare_time', 'holiday', 'type', 'style', 'serves',\
    'suitable_with', 'available','capacity', 'serves',  'price_range']].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2092 entries, 0 to 2091
Data columns (total 16 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   name               2092 non-null   object
 1   address            2092 non-null   object
 2   district           2092 non-null   object
 3   opening_time       2092 non-null   object
 4   suitable_time      2088 non-null   object
 5   last_pick_up_time  1233 non-null   object
 6   prepare_time       1888 non-null   object
 7   holiday            1888 non-null   object
 8   type               2092 non-null   object
 9   style              1918 non-null   object
 10  serves             1970 non-null   object
 11  suitable_with      1889 non-null   object
 12  available          2092 non-null   object
 13  capacity           1876 non-null   object
 14  serves             1970 non-null   object
 15  price_range        132 non-null    object
dtypes: object(16)
memory usage: 277.8+ KB


# Khám phá dữ liệu 2 

File dữ liệu Restaurant_information.txt

In [171]:
pd.set_option('display.max_colwidth', 50)
store_info_df_2 = pd.read_csv('Restaurant_information.txt',delimiter='\t',index_col='id')
store_info_df_2.head()

Link  \
id                                                      
0                /ho-chi-minh/i-love-kem-phan-huy-ich   
1              /ho-chi-minh/i-love-kem-nguyen-anh-thu   
2             /ho-chi-minh/i-love-kem-dinh-tien-hoang   
3   /ho-chi-minh/banh-bong-lan-trung-muoi-to-rang-...   
4             /ho-chi-minh/cua-vang-hai-san-tuoi-song   

                                                 Name  average_score  \
id                                                                     
0                           I Love Kem - Phan Huy Ích            NaN   
1                         I Love Kem - Nguyễn Ảnh Thủ            8.7   
2                        I Love Kem - Đinh Tiên Hoàng            7.5   
3   Bánh Bông Lan Trứng Muối - Tờ Rang - Mai Lão Bạng            8.2   
4                        Cua Vàng - Hải Sản Tươi Sống            7.2   

   space_score position_score quality_score serve_score price_score  \
id                                                                    
0           --             --            --          --          --   
1          9.5            9.5           9.0         9.3         6.0   
2          6.8            8.1           7.6         7.6         7.6   
3          7.2            7.8           9.1         8.5         8.5   
4          7.6            7.2           7.4         6.8         7.0   

                             address       district  ...  \
id                                                   ...   
0            206 Phan Huy Ích, P. 12    Quận Gò Vấp  ...   
1   53 Nguyễn Ảnh Thủ, P. Hiệp Thành        Quận 12  ...   
2   207 Đinh Tiên Hoàng, P. Tân Định         Quận 1  ...   
3             25 Mai Lão Bạng, P. 13  Quận Tân Bình  ...   
4                 16 Đất Thánh, P. 6  Quận Tân Bình  ...   

                             suitable_time  last_pickup_time  \
id                                                             
0   Buổi sáng,Buổi  trưa,Buổi xế,Buổi  tối             22:45   
1   Buổi sáng,Buổi  trưa,Buổi xế,Buổi  tối             22:45   
2             Buổi  trưa,Buổi xế,Buổi  tối             21:30   
3   Buổi sáng,Buổi  trưa,Buổi xế,Buổi  tối          07:30 PM   
4   Buổi sáng,Buổi  trưa,Buổi xế,Buổi  tối          10:00 PM   

           prepare_time                holiday          type       capacity  \
id                                                                            
0    Khoảng 7 - 10 phút  Các ngày lễ trong năm  Café/Dessert   50 người lớn   
1    Khoảng 7 - 10 phút  Các ngày lễ trong năm  Café/Dessert   50 người lớn   
2    Khoảng 5 - 10 phút  Các ngày lễ trong năm  Café/Dessert   30 người lớn   
3                   NaN                    NaN   Shop Online            NaN   
4   Khoảng 10 - 15 phút  Các ngày lễ trong năm      Nhà hàng  300 người lớn   

       style                                suitable_with  \
id                                                          
0   Món Việt           Hẹn hò,Họp nhóm,Takeaway - Mang về   
1   Món Việt           Hẹn hò,Họp nhóm,Takeaway - Mang về   
2   Món Việt           Hẹn hò,Họp nhóm,Takeaway - Mang về   
3   Món Việt                                          NaN   
4   Món Việt  Ăn gia đình,Hẹn hò,Uống bia - Nhậu,Họp nhóm   

                        serves  \
id                               
0                   Café - Kem   
1   Café - Kem,Ăn vặt - Ăn nhẹ   
2                   Café - Kem   
3                   Bánh - Kẹo   
4                      Hải sản   

                                            available  
id                                                     
0   ['Có wifi ', 'Có giao hàng', 'Có máy lạnh & đi...  
1   ['Có wifi ', 'Có giao hàng', 'Có máy lạnh & đi...  
2   ['Có wifi ', 'Có giao hàng ', 'Có máy lạnh & đ...  
3   ['Có giao hàng ', 'Cho mua về ', 'Nên đặt trướ...  
4   ['Có wifi ', 'Có giao hàng', 'Có máy lạnh & đi...  

[5 rows x 25 columns]

## Thông tin chung về dữ liệu 

In [181]:
# Số dòng và cột
print('Số dòng và số cột: {0} {1}'.format(store_info_df_2.shape[0],store_info_df_2.shape[1]))
# Có dòng bị lặp không
print('Số dòng bị lặp: ', store_info_df_2.duplicated().sum())

Số dòng và số cột: 2180 25
Số dòng bị lặp:  0


In [182]:
store_info_df_2.dtypes

Link                 object
Name                 object
average_score       float64
space_score         float32
position_score      float32
quality_score       float32
serve_score         float32
price_score         float32
address              object
district             object
views                object
review_count          int64
price_range          object
open hours           object
close hours          object
suitable_time        object
last_pickup_time     object
prepare_time         object
holiday              object
type                 object
capacity             object
style                object
suitable_with        object
serves               object
available            object
dtype: object

Kiểm tra thông tin của các cột: ` 'space_score', 'position_score', 'quality_score', 'serve_score', 'price_score', 'views' `

In [183]:
store_info_df_2[['space_score', 'position_score', 'quality_score', 'serve_score', 'price_score', 'views']].describe()

space_score  position_score  quality_score  serve_score  price_score
count  1961.000000     1961.000000    1961.000000  1961.000000  1961.000000
mean      7.047272        7.294646       7.260428     7.013412     6.976441
std       1.171125        1.004392       1.267190     1.303124     1.196861
min       1.000000        1.000000       1.000000     1.000000     1.000000
25%       6.300000        6.800000       6.600000     6.300000     6.300000
50%       7.000000        7.300000       7.300000     7.000000     7.000000
75%       7.800000        7.800000       8.000000     7.800000     7.600000
max      10.000000       10.000000      10.000000    10.000000    10.000000

In [175]:
print(store_info_df_2['space_score'].unique())
store_info_df_2['space_score'] = store_info_df_2['space_score'].str.replace('--','nan',regex=False).astype('float32')
store_info_df_2['space_score'].unique()

['--' '9.5' '6.8' '7.2' '7.6' '6.7' '8.3' '9.0' '5.6' '8.5' '6.6' '6.3'
 '6.0' '6.5' '7.5' '7.9' '6.9' '5.8' '5.7' '4.8' '7.1' '7.4' '3.5' '8.6'
 '8.1' '5.9' '8.7' '8.2' '7.7' '9.1' '8.4' '7.3' '7.0' '9.8' '4.2' '3.0'
 '5.1' '7.8' '6.2' '6.1' '6.4' '8.0' '5.0' '8.8' '9.6' '10.0' '5.3' '5.5'
 '4.4' '9.7' '5.4' '9.2' '3.1' '9.3' '8.9' '4.5' '5.2' '4.9' '9.4' '9.9'
 '4.1' '4.7' '2.0' '4.0' '2.3' '3.6' '3.8' '3.7' '1.0' '4.6' '4.3' '1.1'
 '3.4' '2.8']


array([ nan,  9.5,  6.8,  7.2,  7.6,  6.7,  8.3,  9. ,  5.6,  8.5,  6.6,
        6.3,  6. ,  6.5,  7.5,  7.9,  6.9,  5.8,  5.7,  4.8,  7.1,  7.4,
        3.5,  8.6,  8.1,  5.9,  8.7,  8.2,  7.7,  9.1,  8.4,  7.3,  7. ,
        9.8,  4.2,  3. ,  5.1,  7.8,  6.2,  6.1,  6.4,  8. ,  5. ,  8.8,
        9.6, 10. ,  5.3,  5.5,  4.4,  9.7,  5.4,  9.2,  3.1,  9.3,  8.9,
        4.5,  5.2,  4.9,  9.4,  9.9,  4.1,  4.7,  2. ,  4. ,  2.3,  3.6,
        3.8,  3.7,  1. ,  4.6,  4.3,  1.1,  3.4,  2.8], dtype=float32)

In [176]:
print(store_info_df_2['position_score'].unique())
store_info_df_2['position_score'] = store_info_df_2['position_score'].str.replace('--','nan',regex=False).astype('float32')
store_info_df_2['position_score'].unique()

['--' '9.5' '8.1' '7.8' '7.2' '6.6' '8.2' '9.4' '6.5' '7.3' '6.4' '6.3'
 '6.2' '6.8' '7.4' '7.0' '8.5' '5.3' '7.6' '6.7' '5.0' '6.9' '7.1' '8.0'
 '8.4' '8.3' '8.7' '9.0' '6.0' '9.8' '7.5' '5.8' '3.0' '5.9' '7.9' '9.6'
 '5.7' '7.7' '6.1' '10.0' '4.4' '8.6' '9.3' '3.4' '9.7' '9.2' '5.4' '9.1'
 '5.6' '3.9' '8.9' '8.8' '4.7' '2.7' '4.9' '4.0' '5.2' '1.0' '9.9' '4.3'
 '5.5' '4.8' '1.3' '1.1' '4.5' '5.1' '3.3']


array([ nan,  9.5,  8.1,  7.8,  7.2,  6.6,  8.2,  9.4,  6.5,  7.3,  6.4,
        6.3,  6.2,  6.8,  7.4,  7. ,  8.5,  5.3,  7.6,  6.7,  5. ,  6.9,
        7.1,  8. ,  8.4,  8.3,  8.7,  9. ,  6. ,  9.8,  7.5,  5.8,  3. ,
        5.9,  7.9,  9.6,  5.7,  7.7,  6.1, 10. ,  4.4,  8.6,  9.3,  3.4,
        9.7,  9.2,  5.4,  9.1,  5.6,  3.9,  8.9,  8.8,  4.7,  2.7,  4.9,
        4. ,  5.2,  1. ,  9.9,  4.3,  5.5,  4.8,  1.3,  1.1,  4.5,  5.1,
        3.3], dtype=float32)

In [177]:
print(store_info_df_2['quality_score'].unique())
store_info_df_2['quality_score'] = store_info_df_2['quality_score'].str.replace('--','nan',regex=False).astype('float32')
store_info_df_2['quality_score'].unique()

['--' '9.0' '7.6' '9.1' '7.4' '8.1' '7.2' '8.2' '6.6' '7.0' '6.5' '6.8'
 '6.3' '6.9' '8.9' '8.6' '6.1' '4.2' '7.5' '7.1' '5.0' '8.8' '6.7' '6.4'
 '7.8' '8.5' '7.9' '9.4' '8.3' '6.2' '9.8' '5.3' '1.0' '6.0' '4.7' '9.9'
 '7.7' '7.3' '8.0' '5.1' '9.6' '9.7' '9.5' '8.4' '5.9' '4.5' '5.6' '1.5'
 '3.6' '10.0' '8.7' '5.8' '5.7' '4.6' '9.3' '5.5' '4.1' '4.9' '3.8' '3.4'
 '2.0' '5.4' '4.3' '2.1' '4.0' '4.4' '9.2' '4.8' '3.0' '3.9' '5.2' '3.3'
 '3.7']


array([ nan,  9. ,  7.6,  9.1,  7.4,  8.1,  7.2,  8.2,  6.6,  7. ,  6.5,
        6.8,  6.3,  6.9,  8.9,  8.6,  6.1,  4.2,  7.5,  7.1,  5. ,  8.8,
        6.7,  6.4,  7.8,  8.5,  7.9,  9.4,  8.3,  6.2,  9.8,  5.3,  1. ,
        6. ,  4.7,  9.9,  7.7,  7.3,  8. ,  5.1,  9.6,  9.7,  9.5,  8.4,
        5.9,  4.5,  5.6,  1.5,  3.6, 10. ,  8.7,  5.8,  5.7,  4.6,  9.3,
        5.5,  4.1,  4.9,  3.8,  3.4,  2. ,  5.4,  4.3,  2.1,  4. ,  4.4,
        9.2,  4.8,  3. ,  3.9,  5.2,  3.3,  3.7], dtype=float32)

In [178]:
print(store_info_df_2['serve_score'].unique())
store_info_df_2['serve_score'] = store_info_df_2['serve_score'].str.replace('--','nan',regex=False).astype('float32')
store_info_df_2['serve_score'].unique()

['--' '9.3' '7.6' '8.5' '6.8' '6.9' '8.0' '9.8' '6.7' '6.2' '9.0' '6.5'
 '7.0' '8.6' '7.2' '6.4' '5.8' '8.3' '5.4' '4.8' '7.5' '7.4' '8.8' '7.1'
 '5.0' '8.7' '7.9' '8.4' '9.2' '6.3' '4.0' '6.0' '5.2' '1.0' '5.7' '4.2'
 '5.9' '7.7' '9.9' '4.3' '5.1' '7.3' '9.5' '9.6' '7.8' '6.6' '6.1' '4.5'
 '5.5' '3.0' '4.4' '10.0' '8.9' '8.1' '9.7' '8.2' '5.3' '9.4' '5.6' '4.7'
 '4.9' '3.8' '9.1' '2.8' '3.3' '4.1' '2.7' '3.9' '4.6' '3.5' '1.1' '3.7'
 '2.1' '2.3' '2.6' '2.5' '1.8']


array([ nan,  9.3,  7.6,  8.5,  6.8,  6.9,  8. ,  9.8,  6.7,  6.2,  9. ,
        6.5,  7. ,  8.6,  7.2,  6.4,  5.8,  8.3,  5.4,  4.8,  7.5,  7.4,
        8.8,  7.1,  5. ,  8.7,  7.9,  8.4,  9.2,  6.3,  4. ,  6. ,  5.2,
        1. ,  5.7,  4.2,  5.9,  7.7,  9.9,  4.3,  5.1,  7.3,  9.5,  9.6,
        7.8,  6.6,  6.1,  4.5,  5.5,  3. ,  4.4, 10. ,  8.9,  8.1,  9.7,
        8.2,  5.3,  9.4,  5.6,  4.7,  4.9,  3.8,  9.1,  2.8,  3.3,  4.1,
        2.7,  3.9,  4.6,  3.5,  1.1,  3.7,  2.1,  2.3,  2.6,  2.5,  1.8],
      dtype=float32)

In [179]:
print(store_info_df_2['price_score'].unique())
store_info_df_2['price_score'] = store_info_df_2['price_score'].str.replace('--','nan',regex=False).astype('float32')
store_info_df_2['price_score'].unique()

['--' '6.0' '7.6' '8.5' '7.0' '7.3' '7.8' '8.0' '7.7' '7.1' '6.9' '7.5'
 '6.5' '8.3' '6.8' '6.6' '6.3' '5.8' '7.2' '6.7' '5.4' '8.4' '5.9' '5.0'
 '8.1' '6.1' '5.1' '8.9' '9.2' '6.4' '5.2' '4.3' '9.8' '5.3' '4.0' '4.5'
 '7.9' '8.7' '9.6' '6.2' '7.4' '4.8' '9.0' '3.7' '9.7' '8.2' '9.5' '3.0'
 '3.6' '10.0' '8.8' '5.5' '5.7' '9.1' '5.6' '9.4' '3.2' '4.2' '4.7' '9.3'
 '8.6' '1.0' '3.1' '4.4' '4.9' '9.9' '3.4' '4.6' '2.3' '3.3' '1.1' '3.9'
 '3.5']


array([ nan,  6. ,  7.6,  8.5,  7. ,  7.3,  7.8,  8. ,  7.7,  7.1,  6.9,
        7.5,  6.5,  8.3,  6.8,  6.6,  6.3,  5.8,  7.2,  6.7,  5.4,  8.4,
        5.9,  5. ,  8.1,  6.1,  5.1,  8.9,  9.2,  6.4,  5.2,  4.3,  9.8,
        5.3,  4. ,  4.5,  7.9,  8.7,  9.6,  6.2,  7.4,  4.8,  9. ,  3.7,
        9.7,  8.2,  9.5,  3. ,  3.6, 10. ,  8.8,  5.5,  5.7,  9.1,  5.6,
        9.4,  3.2,  4.2,  4.7,  9.3,  8.6,  1. ,  3.1,  4.4,  4.9,  9.9,
        3.4,  4.6,  2.3,  3.3,  1.1,  3.9,  3.5], dtype=float32)

In [184]:
a = store_info_df_2['views'].str.extract(r'(\d+.\d+)K')
b = store_info_df_2['views'].str.extract(r'(\d+)')
a = a + '00'
a = a.fillna(b)
store_info_df_2['views'] = a[0].str.replace('.','',regex=False)
store_info_df_2['views'] = store_info_df_2['views'].astype('int64')

In [185]:
store_info_df_2[['space_score', 'position_score', 'quality_score', 'serve_score', 'price_score', 'views']].describe()

space_score  position_score  quality_score  serve_score  price_score  \
count  1961.000000     1961.000000    1961.000000  1961.000000  1961.000000   
mean      7.047272        7.294646       7.260428     7.013412     6.976441   
std       1.171125        1.004392       1.267190     1.303124     1.196861   
min       1.000000        1.000000       1.000000     1.000000     1.000000   
25%       6.300000        6.800000       6.600000     6.300000     6.300000   
50%       7.000000        7.300000       7.300000     7.000000     7.000000   
75%       7.800000        7.800000       8.000000     7.800000     7.600000   
max      10.000000       10.000000      10.000000    10.000000    10.000000   

               views  
count    2180.000000  
mean    18591.607798  
std     22166.752909  
min         7.000000  
25%      1400.000000  
50%      9250.000000  
75%     29225.000000  
max    137400.000000

In [186]:
store_info_df_2.dtypes

Link                 object
Name                 object
average_score       float64
space_score         float32
position_score      float32
quality_score       float32
serve_score         float32
price_score         float32
address              object
district             object
views                 int64
review_count          int64
price_range          object
open hours           object
close hours          object
suitable_time        object
last_pickup_time     object
prepare_time         object
holiday              object
type                 object
capacity             object
style                object
suitable_with        object
serves               object
available            object
dtype: object

- Các cột dữ liệu dạng số là: 

` 'review_count', 'average_score', 'space_score', 'position_score', 'quality_score','serve_score','price_score','views'`

- Các cột dữ liệu dạng phân loại là:

`'Name', 'address', 'district', 'open hours', 'close_hours', 'suitable_time', 'last_pick_up_time',  'prepare_time', 'holiday', 'type', 'style', 'serves', 'suitable_with', 'available', 'capacity', 'price_range' `

Kiểm tra lại thông tin của những cột dữ liệu trên

In [170]:
store_info_df_2[['review_count', 'average_score', 'space_score', 'position_score', 'quality_score',\
                 'serve_score','price_score','views']].describe()

review_count  average_score  space_score  position_score  \
count   2180.000000    1961.000000  1961.000000     1961.000000   
mean      47.194037       7.115247     7.047272        7.294646   
std       38.911656       1.089733     1.171125        1.004392   
min        0.000000       1.000000     1.000000        1.000000   
25%       11.000000       6.500000     6.300000        6.800000   
50%       47.000000       7.100000     7.000000        7.300000   
75%       70.000000       7.700000     7.800000        7.800000   
max      465.000000      10.000000    10.000000       10.000000   

       quality_score  serve_score  price_score          views  
count    1961.000000  1961.000000  1961.000000    2180.000000  
mean        7.260428     7.013412     6.976441   18591.607798  
std         1.267190     1.303124     1.196861   22166.752909  
min         1.000000     1.000000     1.000000       7.000000  
25%         6.600000     6.300000     6.300000    1400.000000  
50%         7.300000     7.000000     7.000000    9250.000000  
75%         8.000000     7.800000     7.600000   29225.000000  
max        10.000000    10.000000    10.000000  137400.000000

In [29]:
store_info_df_2[['Name', 'address', 'district', 'open hours', 'close hours', 'suitable_time', 'last_pickup_time',\
'prepare_time','holiday', 'type', 'style', 'serves',\
'suitable_with', 'available', 'capacity', 'price_range']].describe()

Name                        address  \
count                                    2180                           2180   
unique                                   2176                           2137   
top     Vegeta - Ẩm Thực Chay - Điện Biên Phủ  3C Tôn Đức Thắng, P. Bến Nghé   
freq                                        2                              6   

       district open hours close hours  \
count      2180       2152        2152   
unique       23         59          50   
top      Quận 1     09:00        22:00   
freq        446        339         778   

                                 suitable_time last_pickup_time  \
count                                     2178             1194   
unique                                      21               70   
top     Buổi sáng,Buổi  trưa,Buổi xế,Buổi  tối         09:30 PM   
freq                                      1310              359   

              prepare_time                holiday     type     style  \
count                 1932                   1932     2180      1991   
unique                  45                     43       61       103   
top     Khoảng 5 - 10 phút  Các ngày lễ trong năm  Quán ăn  Món Việt   
freq                   884                   1824      760      1215   

            serves                       suitable_with  \
count         2075                                1900   
unique        1171                                 527   
top     Bánh - Kẹo  Hẹn hò,Họp nhóm,Takeaway - Mang về   
freq            53                                 138   

                                                available      capacity  \
count                                                2180          1854   
unique                                                622            66   
top     ['Có wifi', 'Có giao hàng', 'Có máy lạnh & điề...  30 người lớn   
freq                                                  137           410   

              price_range  
count                2174  
unique                736  
top     20.000đ - 50.000đ  
freq                   58

# Khám phá dữ liệu 3 

File dữ liệu store_info_updated_1_10.txt

In [136]:
store_info_df_3 = pd.read_csv('store_info_updated_1_10.txt',delimiter=',',index_col='idNum')
store_info_df_3.head()

_id                                  name  \
idNum                                                                   
0      6337f7ea41d0d217e52a06d2          TukTuk Thai Bistro - Hoa Mai   
1      6337f7ef41d0d217e52a06d3                   Papaxốt - Phạm Hùng   
2      6337f7f841d0d217e52a06d4     Trà Sữa BoBaPop - Lãnh Binh Thăng   
3      6337f7fb41d0d217e52a06d5  Lẩu Gà Ớt Hiểm 109 - Nguyễn Thị Thập   
4      6337f80841d0d217e52a06d6                            Hawaii Kìa   

      review_count                              address        district  \
idNum                                                                     
0               55                           13 Hoa Mai  Quận Phú Nhuận   
1              118                   8A Phạm Hùng, P. 4          Quận 8   
2               70           376 Lãnh Binh Thăng, P. 11         Quận 11   
3               13  158A Nguyễn Thị Thập, P. Bình Thuận          Quận 7   
4              278                      401 Sư Vạn Hạnh         Quận 10   

      average_score   views food_list comment_list  quality_score  ...  \
idNum                                                              ...   
0               7.5   11.4K        []           []            7.9  ...   
1               7.3   11.3K        []           []            7.7  ...   
2               6.2   12.0K        []           []            6.3  ...   
3               6.2    2.5K        []           []            5.5  ...   
4               7.0  174.4K        []           []            6.9  ...   

            capacity              style  \
idNum                                     
0       80 người lớn  Món Á ,  Món Thái   
1      100 người lớn  Món Á ,  Món Nhật   
2       25 người lớn  Món Á ,  Đài Loan   
3       50 người lớn           Món Việt   
4       50 người lớn           Món Việt   

                                           suitable_with  \
idNum                                                      
0                 Ăn gia đình ,  Họp nhóm ,  Ăn Fastfood   
1      Đãi tiệc ,  Ăn gia đình ,  Họp nhóm ,  Tiếp khách   
2      Hẹn hò ,  Họp nhóm ,  Thư giãn ,  Takeaway - M...   
3             Ăn gia đình ,  Uống bia - Nhậu ,  Họp nhóm   
4                         Hẹn hò ,  Họp nhóm ,  Thư giãn   

                                                  serves  \
idNum                                                      
0      Thịt Gà ,  Bánh mì ,  Tôm ,  Đậu hũ ,  Bún ,  ...   
1                    Thịt Bò ,  Bún ,  Mì ,  Miến ,  Cơm   
2      Café - Kem ,  Sinh tố - Nước ép ,  Thức uống ,...   
3      Thịt Gà ,  Xôi ,  Bánh bao ,  Mì ,  Lẩu ,  Bán...   
4      Trà ,  Café - Kem ,  Sinh tố - Nước ép ,  Thức...   

                                               available  \
idNum                                                      
0      ['Có wifi', 'Có máy lạnh & điều hòa', 'Nên đặt...   
1      ['Có wifi', 'Nên đặt trước', 'Giữ xe máy miễn ...   
2      ['Có wifi', 'Có giao hàng', 'Có máy lạnh & điề...   
3           ['Có wifi', 'Có giao hàng', 'Nên đặt trước']   
4       ['Có wifi', 'Cho mua về', 'Giữ xe máy miễn phí']   

                                            parking_lots parking_lots_ammount  \
idNum                                                                           
0                                                     []                    0   
1                                                     []                    0   
2                                                     []                    0   
3                                                     []                    0   
4      [{'name': 'Bãi Giữ Xe Đường Cao Thắng', 'addre...                    1   

                                                    link last_pick_up_time  \
idNum                                                                        
0      https://www.foody.vn/ho-chi-minh/tuktuk-thai-b...               NaN   
1      https://www.foody.vn/ho-chi-minh/papaxot-pham-...          09:30 PM   
2      https://www.foody.v

## Thông tin chung về dữ liệu

In [137]:
# Số dòng và cột
print('Số dòng và số cột: {0} {1}'.format(store_info_df_3.shape[0],store_info_df_3.shape[1]))
# Có dòng bị lặp không
print('Số dòng bị lặp: ', store_info_df_3.duplicated().sum())

Số dòng và số cột: 2092 31
Số dòng bị lặp:  0


In [138]:
store_info_df_3.dtypes

_id                      object
name                     object
review_count             object
address                  object
district                 object
average_score            object
views                    object
food_list                object
comment_list             object
quality_score           float64
space_score             float64
position_score          float64
serve_score             float64
price_score             float64
food_link                object
price_range              object
opening_time             object
suitable_time            object
prepare_time             object
holiday                  object
type                     object
capacity                 object
style                    object
suitable_with            object
serves                   object
available                object
parking_lots             object
parking_lots_ammount      int64
link                     object
last_pick_up_time        object
website                  object
dtype: o

In [139]:
store_info_df_3[['review_count', 'average_score','views']].describe()

review_count average_score views
count          2092          2092  2092
unique          466            73  1060
top               0           _._  1.3K
freq            160           160    18

In [140]:
print(store_info_df_3['review_count'].unique())
store_info_df_3['review_count'] = store_info_df_3['review_count'].str.replace(',','').astype('int32')
print(store_info_df_3['review_count'].unique())

['55' '118' '70' '13' '278' '4' '197' '116' '139' '73' '96' '1,366' '33'
 '104' '11' '88' '15' '3' '38' '6' '81' '66' '48' '407' '145' '218' '359'
 '551' '8' '107' '5' '56' '19' '177' '172' '217' '27' '1' '214' '187' '0'
 '10' '25' '255' '52' '113' '227' '397' '466' '112' '161' '434' '189' '94'
 '128' '252' '166' '119' '375' '146' '24' '17' '220' '287' '276' '790'
 '124' '92' '872' '272' '228' '138' '129' '223' '152' '131' '59' '299'
 '234' '126' '2' '82' '628' '263' '141' '37' '360' '110' '285' '308' '125'
 '1,089' '171' '385' '167' '31' '60' '100' '30' '45' '336' '72' '179' '68'
 '130' '51' '461' '245' '84' '546' '310' '36' '137' '95' '9' '330' '76'
 '782' '58' '304' '21' '78' '326' '122' '248' '47' '144' '281' '74' '208'
 '120' '173' '43' '91' '14' '114' '89' '80' '199' '162' '229' '253' '524'
 '39' '157' '69' '42' '577' '105' '340' '127' '44' '79' '149' '111' '86'
 '159' '184' '20' '143' '415' '133' '26' '192' '77' '12' '231' '553' '35'
 '63' '16' '150' '103' '550' '115' '53' '54' 

In [141]:
print(store_info_df_3['average_score'].unique())
store_info_df_3['average_score'] = store_info_df_3['average_score'].str.replace('_._','nan',regex=False).astype('float32')
store_info_df_3['average_score'].unique()

['7.5' '7.3' '6.2' '7.0' '8.0' '7.8' '6.6' '8.8' '7.1' '7.7' '7.2' '5.9'
 '6.5' '7.9' '6.4' '6.8' '8.4' '8.6' '9.2' '6.7' '8.5' '5.8' '6.3' '8.3'
 '10.0' '_._' '6.0' '9.7' '5.4' '7.4' '9.8' '5.5' '6.9' '7.6' '8.1' '9.0'
 '4.9' '9.9' '5.3' '5.6' '6.1' '8.9' '5.7' '8.7' '8.2' '3.8' '4.6' '5.2'
 '5.0' '3.4' '3.6' '4.7' '9.6' '5.1' '2.2' '9.3' '9.1' '9.4' '4.3' '2.0'
 '4.8' '4.1' '1.6' '4.5' '4.0' '3.3' '3.1' '9.5' '3.2' '1.0' '2.8' '2.4'
 '4.2']


array([ 7.5,  7.3,  6.2,  7. ,  8. ,  7.8,  6.6,  8.8,  7.1,  7.7,  7.2,
        5.9,  6.5,  7.9,  6.4,  6.8,  8.4,  8.6,  9.2,  6.7,  8.5,  5.8,
        6.3,  8.3, 10. ,  nan,  6. ,  9.7,  5.4,  7.4,  9.8,  5.5,  6.9,
        7.6,  8.1,  9. ,  4.9,  9.9,  5.3,  5.6,  6.1,  8.9,  5.7,  8.7,
        8.2,  3.8,  4.6,  5.2,  5. ,  3.4,  3.6,  4.7,  9.6,  5.1,  2.2,
        9.3,  9.1,  9.4,  4.3,  2. ,  4.8,  4.1,  1.6,  4.5,  4. ,  3.3,
        3.1,  9.5,  3.2,  1. ,  2.8,  2.4,  4.2], dtype=float32)

In [143]:
a = store_info_df_3['views'].str.extract(r'(\d+.\d+)K')
b = store_info_df_3['views'].str.extract(r'(\d+)')
a = a + '00'
a = a.fillna(b)
store_info_df_3['views'] = a[0].str.replace('.','',regex=False)
store_info_df_3['views'] = store_info_df_3['views'].astype('int64')

In [144]:
store_info_df_3.describe()

review_count  average_score          views  quality_score  space_score  \
count   2092.000000    1932.000000    2092.000000    2092.000000  2092.000000   
mean     133.770554       7.183955   35931.647228       6.673662     6.574235   
std      166.629203       0.998674   49546.810187       2.474002     2.418716   
min        0.000000       1.000000      10.000000      -1.000000    -1.000000   
25%       19.000000       6.700000    2800.000000       6.600000     6.300000   
50%       89.000000       7.200000   12800.000000       7.300000     7.100000   
75%      179.000000       7.625000   55800.000000       7.900000     7.800000   
max     1513.000000      10.000000  459500.000000      10.000000    10.000000   

       position_score  serve_score  price_score  parking_lots_ammount  
count     2092.000000  2092.000000  2092.000000           2092.000000  
mean         6.765727     6.444503     6.342304              3.824570  
std          2.413706     2.433752     2.339680              6.487033  
min         -1.000000    -1.000000    -1.000000              0.000000  
25%          6.800000     6.200000     6.300000              0.000000  
50%          7.300000     7.000000     6.800000              0.000000  
75%          7.800000     7.600000     7.300000              4.000000  
max         10.000000    10.000000    10.000000             25.000000

# Gộp lại các file dữ liệu sau khi đã tiền xử lý

**Kiểm tra lại thông tin của các DataFrame**

In [188]:
store_info_df.shape

(2092, 29)

In [189]:
store_info_df_2.shape

(2180, 25)

In [190]:
store_info_df_3.shape

(2092, 31)

In [191]:
store_info_df.dtypes

_id                   object
name                  object
review_count           int32
address               object
district              object
average_score        float32
views                  int64
food_list             object
comment_list          object
position_score       float64
space_score          float64
quality_score        float64
serve_score          float64
price_score          float64
food_link             object
opening_time          object
suitable_time         object
last_pick_up_time     object
prepare_time          object
holiday               object
type                  object
capacity              object
style                 object
suitable_with         object
serves                object
available             object
parking_lots          object
website               object
price_range           object
dtype: object

In [192]:
store_info_df_2.dtypes

Link                 object
Name                 object
average_score       float64
space_score         float32
position_score      float32
quality_score       float32
serve_score         float32
price_score         float32
address              object
district             object
views                 int64
review_count          int64
price_range          object
open hours           object
close hours          object
suitable_time        object
last_pickup_time     object
prepare_time         object
holiday              object
type                 object
capacity             object
style                object
suitable_with        object
serves               object
available            object
dtype: object

In [193]:
store_info_df_3.dtypes

_id                      object
name                     object
review_count              int32
address                  object
district                 object
average_score           float32
views                     int64
food_list                object
comment_list             object
quality_score           float64
space_score             float64
position_score          float64
serve_score             float64
price_score             float64
food_link                object
price_range              object
opening_time             object
suitable_time            object
prepare_time             object
holiday                  object
type                     object
capacity                 object
style                    object
suitable_with            object
serves                   object
available                object
parking_lots             object
parking_lots_ammount      int64
link                     object
last_pick_up_time        object
website                  object
dtype: o

In [194]:
store_info_df_2.to_csv('preprocessed_Restaurant_information.txt')
store_info_df_3.to_csv('preprocessed_store_info_updated.txt')

In [195]:
store_info_df_2.describe()

average_score  space_score  position_score  quality_score  serve_score  \
count    1961.000000  1961.000000     1961.000000    1961.000000  1961.000000   
mean        7.115247     7.047272        7.294646       7.260428     7.013412   
std         1.089733     1.171125        1.004392       1.267190     1.303124   
min         1.000000     1.000000        1.000000       1.000000     1.000000   
25%         6.500000     6.300000        6.800000       6.600000     6.300000   
50%         7.100000     7.000000        7.300000       7.300000     7.000000   
75%         7.700000     7.800000        7.800000       8.000000     7.800000   
max        10.000000    10.000000       10.000000      10.000000    10.000000   

       price_score          views  review_count  
count  1961.000000    2180.000000   2180.000000  
mean      6.976441   18591.607798     47.194037  
std       1.196861   22166.752909     38.911656  
min       1.000000       7.000000      0.000000  
25%       6.300000    1400.000000     11.000000  
50%       7.000000    9250.000000     47.000000  
75%       7.600000   29225.000000     70.000000  
max      10.000000  137400.000000    465.000000

In [196]:
store_info_df_3.describe()

review_count  average_score          views  quality_score  space_score  \
count   2092.000000    1932.000000    2092.000000    2092.000000  2092.000000   
mean     133.770554       7.183955   35931.647228       6.673662     6.574235   
std      166.629203       0.998674   49546.810187       2.474002     2.418716   
min        0.000000       1.000000      10.000000      -1.000000    -1.000000   
25%       19.000000       6.700000    2800.000000       6.600000     6.300000   
50%       89.000000       7.200000   12800.000000       7.300000     7.100000   
75%      179.000000       7.625000   55800.000000       7.900000     7.800000   
max     1513.000000      10.000000  459500.000000      10.000000    10.000000   

       position_score  serve_score  price_score  parking_lots_ammount  
count     2092.000000  2092.000000  2092.000000           2092.000000  
mean         6.765727     6.444503     6.342304              3.824570  
std          2.413706     2.433752     2.339680              6.487033  
min         -1.000000    -1.000000    -1.000000              0.000000  
25%          6.800000     6.200000     6.300000              0.000000  
50%          7.300000     7.000000     6.800000              0.000000  
75%          7.800000     7.600000     7.300000              4.000000  
max         10.000000    10.000000    10.000000             25.000000